modules/dag-l1/src/main/scala/org/tessellation/dag/l1/StateChannel.scala

```diff
@@ -120,9 +120,9 @@ class StateChannel[
 
   private val runConsensus: Pipe[F, BlockConsensusInput, FinalBlock] =
     _.evalTap(input => logger.debug(s"Received block consensus input to process: ${input.show}"))
-      .evalMap(
-        new BlockConsensusCell[F](_, blockConsensusContext)
-          .run()
+      .evalMap(blockConsensusInput =>
+        storages.lastSnapshot.getOrdinal
```

Comment:
is it ensured the process starts when the `storages.lastSnapshot` has been set?

Reply:
I don't understand.
cc: @ryle-goehausen

Reply:
I'm confused. Didn't you mention earlier that at this point we must have a snapshot or consensus can't happen?

Reply:
I'm currently refactoring so that an `F[Option[Snapshot]]` is passed instead. The last hurdle is the previous issue with constructing `BlockAcceptanceManager` in `SharedServices`.

Reply:
refactored to pass `F[Option[Snapshot]]` which isn't evaluated until we get into the block validator

Reply:
> I'm confused. Didn't you mention earlier that at this point we must have a snapshot or consensus can't happen?

well, that's my assumption that the consensus cannot run without a snapshot (we don't know tips), but I ask if you confirmed that (I did not). so it should be fine to get the ordinal here and throw if it's not set, but I'd like you to make sure we always have the ordinal in this place. (like - where the check for snapshot is, before this or somewhere after but before starting the round)

Reply:
@marcinwadon I had not confirmed it but after looking at it again with @ryle-goehausen we noticed other issues with getting the ordinal there so instead getting the ordinal isn't evaluated until we're in the validator, if we're getting it from the storage. If it's already available then it's passed in a pure `F`.

Reply:
I think that passing down the lazy evaluated `F` brings race condition issues and I would suggest to re-check the previous implementation to pass the ordinal instead of evaluating the ordinal in the most bottom(children) component.

Reply:
cc: @TheMMaciek @kpudlik